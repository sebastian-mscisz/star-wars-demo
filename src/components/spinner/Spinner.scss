.sw-spinner {
    display: flex;
    height: 100%;
    align-items: center;
}