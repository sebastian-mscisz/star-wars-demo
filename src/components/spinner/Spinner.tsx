import { BarLoader } from 'react-spinners';
import './Spinner.scss';

export const Spinner: React.FC = () => {
    const colors = ['#EB212E', '#2FF924', '#2E67F8', '#871efe'];

    return (
        <div className="sw-spinner">
            <BarLoader
                height={5}
                cssOverride={{ width: '25px', backgroundColor: 'black', borderRadius: '5px' }}
                speedMultiplier={0}
            />
            <BarLoader height={5} cssOverride={{ borderRadius: '5px' }} color={colors[Math.floor(Math.random() * 4)]} />
        </div>
    );
};
