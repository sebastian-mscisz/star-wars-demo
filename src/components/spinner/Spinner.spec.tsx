import { Spinner } from './Spinner';
import { render } from '@testing-library/react';

describe('Spinner', () => {
    beforeEach(() => {
        jest.spyOn(global.Math, 'random').mockReturnValue(1);
    });

    afterEach(() => {
        jest.spyOn(global.Math, 'random').mockRestore();
    });

    it('should render', () => {
        const { asFragment } = render(<Spinner />);
        expect(asFragment()).toMatchSnapshot();
    });
});
