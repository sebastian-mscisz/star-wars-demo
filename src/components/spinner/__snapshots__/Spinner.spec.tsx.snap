// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render 1`] = `
<DocumentFragment>
  <div
    class="sw-spinner"
  >
    <span
      style="display: inherit; position: relative; width: 25px; height: 5px; overflow: hidden; background-color: black; background-clip: padding-box; border-radius: 5px;"
    >
      <span
        style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long Infinitys  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
      />
      <span
        style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short Infinitys Infinitys cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
      />
    </span>
    <span
      style="display: inherit; position: relative; width: 100px; height: 5px; overflow: hidden; background-color: rgba(0, 0, 0, 0.2); background-clip: padding-box; border-radius: 5px;"
    >
      <span
        style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-long 2.1s  cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;"
      />
      <span
        style="position: absolute; height: 5px; overflow: hidden; background-color: rgb(0, 0, 0); background-clip: padding-box; display: block; border-radius: 2px; will-change: left, right; animation-fill-mode: forwards; animation: react-spinners-BarLoader-short 2.1s 1.15s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;"
      />
    </span>
  </div>
</DocumentFragment>
`;
